pagopa/io-app
# fix: [IABT-1443,IOAPPFD0-60] Invalid characters when pasting OFC

## Short description
This PR implements a formatting feature for the organization fiscal code, during a manual payment, which removes blank spaces and invalid characters when typing or pasting something in the related field.

This PR is greatly inspired by the work done in #4170 and simply applies the same logic to the OFC field.

## List of changes proposed in this pull request
- Removes the character limit to allow pasting strings longer than 11 characters from the OFC field;
- Adds a regex that removes spaces and invalid characters in the input's onChangeText method, in order for the state to be updated with the correct data post-filtering;
- Adds a `9999999999` mask which ignores spaces and invalid characters when typing or pasting data;
- Adds a `ofcInputValue` state variable to set the input's state and renames the existing one to `pnnInputValue` to differentiate them;
- Fixes a regression which caused the hairline to not to change color on input validation by using the `overrideBorderColor`prop.

## How to test
- Navigate to `Wallet -> Pay Notice -> Enter Manually;
- Type an invalid `Notice Code` with less than 18 characters or with a >3 digit in the first position and the hairline should be red;
- Type a valid `Notice Code` and the hairline should be green;
- Test the same functionality for the `Creditor Entity Fiscal Code` field which only accepts 11 digits;
- Paste an invalid code which includes characters or spaces in both field. Only allowed characters should be pasted and spaces should be stripped as well, along with extra characters which exceed the allowed limit.
